src/ip/Address.cc

```diff
@@ -404,7 +404,11 @@ Ip::Address::lookupHostIP(const char *s, bool nodns)
         }
         if (maybeIpv4 != nullptr)
             res = maybeIpv4;
-        // else IPv6-only host, let the caller deal with first-IP anyway.
+        else {
+            // no IPv4 addresses (e.g., an IPv6-only host)
+            freeaddrinfo(resHead);
+            return false;
```

Comment:
> I like the last option (combined with fixing IP address parsing API, but that can be done separately, under less pressure), but it requires a lot of work. Am I missing a better alternative?

I did not receive any feedback and posted #1567. It tries to solve the immediate problem without hitting this particular roadblock. If #1567 is accepted, most changes in this PR should still be merged (after adjusting the low-level stuff to avoid conflicts with #1567 approach).

Reply:
The primary problem behind this change request has been addressed by a combination of master/v7 commits 97bbba61a5ff8c5dfdb24fdbad1e8f3f838cb5fd and 51c518d. Those changes were based on "option 3" approach outlines in this change request. I am resolving this change request.